tests/test_pipeline_common.py

```diff
@@ -0,0 +1,235 @@
+import copy
+import importlib
+import os
+import random
+from pathlib import Path
+
+import requests
+
+from .pipelines.test_pipelines_audio_classification import AudioClassificationPipelineTests
+from .pipelines.test_pipelines_feature_extraction import FeatureExtractionPipelineTests
+from .pipelines.test_pipelines_fill_mask import FillMaskPipelineTests
+from .pipelines.test_pipelines_image_classification import ImageClassificationPipelineTests
+from .pipelines.test_pipelines_text_classification import TextClassificationPipelineTests
+from .pipelines.test_pipelines_text_generation import TextGenerationPipelineTests
+
+
+pipeline_test_mapping = {
+    "audio-classification": {"test": AudioClassificationPipelineTests},
+    "feature-extraction": {"test": FeatureExtractionPipelineTests},
+    "fill-mask": {"test": FillMaskPipelineTests},
+    "image-classification": {"test": ImageClassificationPipelineTests},
+    "text-classification": {"test": TextClassificationPipelineTests},
+    "text-generation": {"test": TextGenerationPipelineTests},
+}
+
+for task, task_info in pipeline_test_mapping.items():
+    test = task_info["test"]
+    task_info["mapping"] = {
+        "pt": getattr(test, "model_mapping", None),
+        "tf": getattr(test, "tf_model_mapping", None),
+    }
+
+
+# Download tiny model summary (used to avoid requesting from Hub too many times)
+url = "https://huggingface.co/datasets/hf-internal-testing/tiny-random-model-summary/raw/main/processor_classes.json"
+tiny_model_summary = requests.get(url).json()
+
+PATH_TO_TRANSFORMERS = os.path.join(Path(__file__).parent.parent, "src/transformers")
+
+
+# Dynamically import the Transformers module to grab the attribute classes of the processor form their names.
+spec = importlib.util.spec_from_file_location(
+    "transformers",
+    os.path.join(PATH_TO_TRANSFORMERS, "__init__.py"),
+    submodule_search_locations=[PATH_TO_TRANSFORMERS],
+)
+transformers_module = spec.loader.load_module()
+
+
+class PipelineTesterMixin:
+    model_tester = None
+    supported_frameworks = ["pt", "tf"]
+
+    @property
+    def framework(self):
+        framework = None
+        if "ModelTesterMixin" in set(x.__name__ for x in self.__class__.__bases__):
+            framework = "pt"
+        elif "TFModelTesterMixin" in set(x.__name__ for x in self.__class__.__bases__):
+            framework = "tf"
+        return framework
```

Comment:
This is used to infer if the tests is for PT or TF (by using the parent class of the test cases).